File: modelo/docente.cs
using System.Data;
using System;
public class docente : oObjeto{

public string Nombre { get; set; }
public string Apellido { get; set; }
public string  Dni { get; set; }
public string Correo{get;set;}
public string Telefono{get; set;}
//foreign key
public int Idusuario{get; set;}
public int Iddomicilio{get; set;}
public int Legajo{get; set;}
public docente(){}
public docente( DataRow dr){
    this.Id = Convert.ToInt32(dr["id"]);
    this.Nombre = dr["nombre"].ToString();
    this.Apellido = dr["apellido"].ToString();
    this.Dni = dr["dni"].ToString();
    this.Correo=dr["correo"].ToString();
    this.Telefono=dr["telefono"].ToString();
    this.Idusuario=Convert.ToInt32(dr["idusuario"]); 
    this.Iddomicilio=Convert.ToInt32(dr["iddomicilio"]);
    this.Legajo= Convert.ToInt32(dr["legajo"]);
}






}

File: ClientApp/src/app/clases/docente.ts
export class docente{
    id : number;
    nombre : string;
    apellido : string;
    dni : string;
    correo: string;
    telefono: string;
    idusuario: number;
    iddomicilio: number;
    legajo: number;
    
    constructor(id: string, nombre:string, apellido:string, dni:string, correo:string, telefono:string, idusuario:string, iddomicilio:string, legajo:string){
        //el orden de os parametros afecta la vista de los input en el modal
        this.id = +id; // formateo a number el string // parseInt(valor)
        this.nombre = nombre;
        this.apellido =  apellido;
        this.dni = dni;
        this.correo=correo;
        this.telefono=telefono;
        this.idusuario=+idusuario; 
        this.iddomicilio=+iddomicilio;
        this.legajo=+legajo;
    }
    
  
}
        